common/persistence/wrappers/sampled/visibility_manager_test.go

```diff
@@ -0,0 +1,344 @@
+		},
+	})
+	assert.Error(t, err, "second call should fail since underlying call should be blocked by rate limiter")
+}
+
+func TestVisibilityManager_ListClosedWorkflowExecutionsByType(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockedManager := persistence.NewMockVisibilityManager(ctrl)
+
+	testDomain := "domain1"
+
+	m := NewVisibilityManager(mockedManager, Params{
+		Config:       &Config{},
+		MetricClient: metrics.NewNoopMetricsClient(),
+		Logger:       testlogger.New(t),
+		TimeSource:   clock.NewMockedTimeSource(),
+		RateLimiterFactoryFunc: rateLimiterStubFunc(map[string]tokenbucket.PriorityTokenBucket{
+			testDomain: &tokenBucketFactoryStub{tokens: map[int]int{0: 1}},
+		}),
+	})
+
+	mockedManager.EXPECT().ListClosedWorkflowExecutionsByType(gomock.Any(), gomock.Any()).Return(&persistence.ListWorkflowExecutionsResponse{}, nil).Times(1)
+
+	_, err := m.ListClosedWorkflowExecutionsByType(context.Background(), &persistence.ListWorkflowExecutionsByTypeRequest{
+		ListWorkflowExecutionsRequest: persistence.ListWorkflowExecutionsRequest{
+			Domain: testDomain,
+		},
+	})
+	assert.NoError(t, err, "first call should succeed")
+
+	_, err = m.ListClosedWorkflowExecutionsByType(context.Background(), &persistence.ListWorkflowExecutionsByTypeRequest{
+		ListWorkflowExecutionsRequest: persistence.ListWorkflowExecutionsRequest{
+			Domain: testDomain,
+		},
+	})
+	assert.Error(t, err, "second call should fail since underlying call should be blocked by rate limiter")
+}
+
+func TestVisibilityManager_ListOpenWorkflowExecutionsByWorkflowID(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockedManager := persistence.NewMockVisibilityManager(ctrl)
+
+	testDomain := "domain1"
+
+	m := NewVisibilityManager(mockedManager, Params{
+		Config:       &Config{},
+		MetricClient: metrics.NewNoopMetricsClient(),
+		Logger:       testlogger.New(t),
+		TimeSource:   clock.NewMockedTimeSource(),
+		RateLimiterFactoryFunc: rateLimiterStubFunc(map[string]tokenbucket.PriorityTokenBucket{
+			testDomain: &tokenBucketFactoryStub{tokens: map[int]int{0: 1}},
+		}),
+	})
+
+	mockedManager.EXPECT().ListOpenWorkflowExecutionsByWorkflowID(gomock.Any(), gomock.Any()).Return(&persistence.ListWorkflowExecutionsResponse{}, nil).Times(1)
+
+	_, err := m.ListOpenWorkflowExecutionsByWorkflowID(context.Background(), &persistence.ListWorkflowExecutionsByWorkflowIDRequest{
+		ListWorkflowExecutionsRequest: persistence.ListWorkflowExecutionsRequest{
+			Domain: testDomain,
+		},
+	})
+	assert.NoError(t, err, "first call should succeed")
+
+	_, err = m.ListOpenWorkflowExecutionsByWorkflowID(context.Background(), &persistence.ListWorkflowExecutionsByWorkflowIDRequest{
+		ListWorkflowExecutionsRequest: persistence.ListWorkflowExecutionsRequest{
+			Domain: testDomain,
+		},
+	})
+	assert.Error(t, err, "second call should fail since underlying call should be blocked by rate limiter")
+}
+
+func TestVisibilityManager_ListClosedWorkflowExecutionsByWorkflowID(t *testing.T) {
```

Comment:
I know you are not fan of table tests but for 99% identical test function bodies I can't see the point of duplicating so much code. any future change will have to be made in N places. maybe I am missing something here. what is the main reasoning for this way of structuring here?

Reply:
Well, in this case table tests are not easy, setup is complicated, though assertion is similar.
I gave it a try and wrote it in table tests style. Give it a look.

Reply:
looks good! thanks